src/rust/engine/graph/src/lib.rs

```diff
@@ -287,12 +289,18 @@ impl<N: Node> InnerGraph<N> {
   /// The Walk will iterate over all nodes that descend from the roots in the direction of
   /// traversal but won't necessarily be in topological order.
   ///
-  fn walk(&self, roots: VecDeque<EntryId>, direction: Direction) -> Walk<'_, N> {
+  fn walk(
+    &self,
+    roots: VecDeque<EntryId>,
+    direction: Direction,
+    stop_at_uncacheable: bool,
```

Comment:
Rather than a boolean, consider making this a predicate, similar to the one for `invalidate_from_roots`?

Reply:
Yeah that would work, not sure I see the real benefit other than sticking to a convention, because most callers of .walk wouldn't want to pass a predicate at all.

Reply:
done.